controlplane/kubeadm/internal/workload_cluster_conditions_test.go

```diff
@@ -980,6 +980,9 @@ func fakeNode(name string, options ...fakeNodeOption) *corev1.Node {
 	p := &corev1.Node{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: name,
+			Labels: map[string]string{
+				labelNodeRoleControlPlane: "",
```

Comment:
This is now necessary because the old implementation of `getControlPlaneNodes` did select all nodes as control plane nodes (I assume because `ctrlclient.MatchingLabels(labels)` is not implemented in fake client)

Reply:
There are tests for it https://github.com/kubernetes-sigs/controller-runtime/blob/f20692b7ce2a9dda7eb298504a9c47aaf5a96544/pkg/client/fake/client_test.go#L314-L324 and it should be supported afaik. 

The code that does the selection is in https://github.com/kubernetes-sigs/controller-runtime/blob/4e7f0c968b90a306fb9ef22f8e92fa36b98b4ef1/pkg/cache/internal/cache_reader.go#L107-L108 — are we not using a cache here perhaps?

Reply:
I'll take a closer look. 

To be clear for the new code everything works fine. I think it's now mostly a question of why didn't we have to set the control plane node label in the old test.

Reply:
Okay everything worked/works as expected.

In the relevant unit test we're injecting the list result: https://github.com/kubernetes-sigs/cluster-api/blob/5026786ee809c5def466049f5befd2a786fbcefa/controlplane/kubeadm/internal/workload_cluster_conditions_test.go#L84-L88

Previously, we didn't have to set the control plane label as the result of the list was automatically the list of control plane nodes. As we're now iterating through the result and checking for the label it became relevant.